sklearn/utils/estimator_checks.py

```diff
@@ -483,8 +484,61 @@ def check_dict_unchanged(name, Estimator):
                               'Estimator changes __dict__ during %s' % method)
 
 
+def _is_public(attr):
+    return not (attr.startswith('_') or attr.endswith('_'))
+
+
+def check_fit_changes_private_attributes_only(name, Estimator):
```

Comment:
hm, can't think about something better than `check_fit_adds_or_changes_underscored_attributes_only`, but that seems strange!